google-beta/resource_bigquery_table.go

```diff
@@ -62,6 +63,182 @@ func resourceBigQueryTable() *schema.Resource {
 				Computed: true,
 			},
 
+			// ExternalDataConfiguration [Optional] Describes the data format,
+			// location, and other properties of a table stored outside of BigQuery.
+			// By defining these properties, the data source can then be queried as
+			// if it were a standard BigQuery table.
+			"external_data_configuration": {
+				Type:     schema.TypeList,
+				Optional: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						// Autodetect : [Required] If true, let BigQuery try to autodetect the
+						// schema and format of the table.
+						"autodetect": {
```

Comment:
I'm not too familiar with Bigquery- would the values autodetected be limited to the `schema` field? I'm wondering if we could omit this entirely. Terraform values being explicit over implicit, and it would have a very hard time reconciling values managed partly by it and partly by Bigquery.

Reply:
I think it will only autodetect the `schema` and `source_format` field, but I have to do some testing. But this actually a very important feature to have since is would be very cumbersome to define schemas inline. It's not uncommon to use CVS's with hundreds or even thousands of columns.

I'll did some testing with both using `schema` and `autodetect` but I'll do some more verification.